manubot/cite/tests/test_csl_item.py

```diff
@@ -179,3 +185,46 @@ def test_csl_item_note_append(input_note, text, dictionary, expected_note):
 def test_csl_item_note_dict(note, dictionary):
     csl_item = CSL_Item(note=note)
     assert csl_item.note_dict == dictionary
+
+
+@pytest.mark.parametrize(
+    ["date", "expected"],
+    [
+        (None, None),
+        ("", None),
+        ("2019", [2019]),
+        ("2019-01", [2019, 1]),
+        ("2019-12", [2019, 12]),
+        ("2019-12-31", [2019, 12, 31]),
+        ("2019-12-99", [2019, 12]),
```

Comment:
Related to the above comment, we could also watch for `2019-12-39` as an invalid day.

Reply:
I am okay with not detecting `2019-12-39` as invalid. We don't gaurantee that we catch invalid dates, although we try to avoid extracting them, we're not actually verifying.